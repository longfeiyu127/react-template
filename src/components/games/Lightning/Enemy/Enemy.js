import React from 'react';
// import Tictactoe from './tictactoe'
// import Fivechess from './fivechess'
import './Enemy.less';

class Enemy extends React.Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    // eslint-disable-next-line
    const { position, enemyType, size } = this.props;
    return (
      <div
        className="g-lig-enemy"
        style={{
          top: `${position.top}rem`,
          left: `${position.left}rem`,
          height: `${size.height}rem`,
          width: `${size.width}rem`
        }}
      >
        {/* style={{fontWeight: (move === stepNumber) ? 900 : 500, cursor: 'pointer'}} */}
        <main className={['enemy-main', enemyType].join(' ')} />
      </div>
    );
  }
}

export default Enemy;
