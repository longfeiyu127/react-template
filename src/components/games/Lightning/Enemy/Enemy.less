.g-lig-enemy{
  position: absolute;
  >.enemy-main {
    height: 100%;
    width: 100%;
  }
  // >.player{
  //   background: url('../../../../../assets/images/games/lightning/bullet/palyerBullet.png') no-repeat center/100% 100%;
  //   animation: flicker 100ms infinite;
  // }
  >.enemyA0{
    background: url('../../../../assets/images/games/lightning/enemy/enemy_typeA.png') no-repeat center/100% 100%;
  }
  >.enemyA1{
    background: url('../../../../assets/images/games/lightning/enemy/enemy_typeA1.png') no-repeat center/100% 100%;
  }
  >.enemyA2{
    background: url('../../../../assets/images/games/lightning/enemy/enemy_typeA2.png') no-repeat center/100% 100%;
  }
  >.enemyC{
    background: url('../../../../assets/images/games/lightning/enemy/enemy_typeC.png') no-repeat center/100% 100%;
  }

  >.explosion{
    background: url('../../../../assets/images/games/lightning/explosion/disp_02.png') no-repeat center/100% 100%;
  }
  // @keyframes flicker {
  //   0%, 100% {
  //     opacity: 0;
  //   }
  //   50% {
  //     opacity: 1;
  //   }
  // }
}
