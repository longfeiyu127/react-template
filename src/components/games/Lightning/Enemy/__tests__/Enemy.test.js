import React from 'react';
import renderer from 'react-test-renderer';
import Enemy from '../Enemy';

describe('<Enemy />', () => {
  it('component render', () => {
    const position = {
      top: 0.1,
      left: 0.1
    };
    const size = {
      height: 0.1,
      width: 0.1
    };
    let enemy = renderer
      .create(<Enemy position={position} size={size} bulletTpey="player" />)
      .toJSON();
    expect(enemy).toMatchSnapshot();

    enemy = renderer
      .create(<Enemy position={position} size={size} bulletTpey="player" />)
      .toJSON();
    expect(enemy).toMatchSnapshot();
  });
});
