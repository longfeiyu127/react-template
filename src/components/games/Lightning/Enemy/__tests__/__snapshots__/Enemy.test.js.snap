// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Enemy /> component render 1`] = `
<div
  className="g-lig-enemy"
  style={
    Object {
      "height": "0.1rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.1rem",
    }
  }
>
  <main
    className="enemy-main "
  />
</div>
`;

exports[`<Enemy /> component render 2`] = `
<div
  className="g-lig-enemy"
  style={
    Object {
      "height": "0.1rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.1rem",
    }
  }
>
  <main
    className="enemy-main "
  />
</div>
`;
